File: estoque-web/src/model/produto.ts
export class Produto {
    id: string;
    nome: string;
    descricao: string;
    preco: number;
    quantidade: number;
    categoria: string;
    fornecedor: string;
    dataCriacao: Date;

    constructor(
        id: string,
        nome: string,
        descricao: string,
        preco: number,
        quantidade: number,
        categoria: string,
        fornecedor: string,
        dataCriacao: Date = new Date()
    ) {
        this.id = id;
        this.nome = nome;
        this.descricao = descricao;
        this.preco = preco;
        this.quantidade = quantidade;
        this.categoria = categoria;
        this.fornecedor = fornecedor;
        this.dataCriacao = dataCriacao;
    }
}

File: estoque/src/main/java/com/example/estoque/model/Produto.java
package com.example.estoque.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "produto")
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    private String nome;

    private String descricao;

    private BigDecimal preco;

    private Integer quantidade;

    private String categoria;

    private UUID fornecedor;

    @Column(name = "data_criacao")
    private LocalDateTime dataCriacao;

    @PrePersist
    protected void onCreate() {
        dataCriacao = LocalDateTime.now();
    }
}